api/uaa/refresh_token.go

```diff
@@ -36,6 +51,7 @@ func (client *Client) RefreshAccessToken(refreshToken string) (RefreshToken, err
 			"Content-Type": {"application/x-www-form-urlencoded"},
 		},
 		Body: body,
+		AddClientAuthorization: true,
 	})
```

Comment:
@mariash - I think you're confusing the HTTP specification with the OAuth specification. You are correct that [RFC2617: HTTP Authentication: Basic and Digest Access Authentication](https://www.rfc-editor.org/rfc/rfc2617.txt) states to simply concatenate with no query escaping, and that is what the Golang HTTP library is correctly doing, since it is an HTTP library, not an OAuth library.

However the [section 2.3.1 of OAuth 2.0 spec](https://tools.ietf.org/html/rfc6749#section-2.3.1) states:

> The client identifier is encoded using the
   "application/x-www-form-urlencoded" encoding algorithm per
   Appendix B, and the encoded value is used as the username; the client
   password is encoded using the same algorithm and used as the
   password.

This means that both the client ID and secret need to be URL encoded before calculating the Basic auth header.